### PR TITLE
Add TypeScript declaration

### PR DESCRIPTION
**Pull request by employee451**
This library is a gem

### CODE REVIEW

**Comment by fubar** [comment]
Haha I didn't think anyone was using it 😄 Thanks for the contribution! PR looks good on first look, and I'll try it out later today.

**Comment by employee451** [comment]
No problem!😄 I enjoy using it

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     * Use this to remove a time range from any existing availabilities.
+     * For example, call this for scheduled appointments or meetings.
+     * @param startDate ISO 8601 string
+     * @param endDate ISO 8601 string
+     */
+    removeAvailability(startDate: string, endDate: string): void
+
+    /**
+     * Returns all availabilities in chronological order.
+     * @param timezone Defaults to '+0000'
```
Comment (fubar) [inline index.d.ts]: Can you add this note to this param:
> Accepts just the timezone offset such as "-05:00" as well as a full time stamp that includes the offset (e.g. "2000-01-01T00:00:00-04:00")

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * Returns true if the given time range falls within any availability.
+     * @param startDate ISO 8601 string
+     * @param endDate ISO 8601 string
+     */
+    isAvailable(startDate: string, endDate: string): boolean
+}
+
+declare module 'availability-schedule' {
+    const AvailabilitySchedule: AvailabilitySchedule
```
Comment (fubar) [inline index.d.ts]: This line should be removed. TS then correctly picks up the class export when imported with:
```ts
import AvailabilitySchedule = require('availability-schedule');
```
(which I'll add to the readme). See https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

**Review by fubar — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (employee451) [reply index.d.ts]: This doesn't work with ES modules: `TS1202: Import assignment cannot be used when targeting ECMAScript modules`

File: index.d.ts (hunk 1 above)
Comment (employee451) [reply index.d.ts]: Done

File: index.d.ts (hunk 2 above)
Comment (fubar) [reply index.d.ts]: How are you importing it? Does this require more changes or all good?

Comment (employee451) [reply index.d.ts]: I think all good

Comment (employee451) [reply index.d.ts]: > ```ts
> import AvailabilitySchedule from 'availability-schedule';
> ```

Comment (fubar) [reply index.d.ts]: Oh nice, I didn't think that works. I'll add a note for the error you got and that the regular "import from" statement works when targeting ES modules.

**Comment by fubar** [comment]
@employee451 released under version 1.1.0. Thanks again for your contribution!

**Pull request merged**